### PR TITLE
Proposal NFC subsystem including nRF-NFCT drivers, nt3h2 driver and sample app.

### PR DESCRIPTION
**Pull request by HesselM**
Hi all,

During the past 2 year we've worked with different boards whom supported NFC.
To ease our lives we made our own subsystem, provided in this PR. 

When I came across #5703 I decided to push our approach to public space.
The proposed setup is upgraded to zephyr-3.4 and includes the latest NFC-drivers from the nrf-sdk and nrfxlib.
It also includes a sample app and supports the nt3h2.

### CODE REVIEW

**Comment by thedjnK** [comment]
This is a non starter, a resounding no, none of this code nor the libraries are apache licensed, please read https://docs.zephyrproject.org/latest/contribute/guidelines.html#contribution-guidelines

**Comment by HesselM** [comment]
@thedjnK : fixed the licenses on all files we created.

**Comment by thedjnK** [comment]
I think you misunderstood my comment:

1. This is not a generic API, this is Nordic's API
2. These files are not apache licensed, the files clearly state they are under Nordic's license
3. Notwithstanding you don't actually have permission to submit them in a PR as I doubt nordic would have given you that permission to post them here

**Review by nashif — CHANGES_REQUESTED** [review]
> @thedjnK : fixed the licenses on all files we created.

this PR has lots of binaries, this is a no-go, nevermind what the license or who the copyright holder is. Please consult the contribution guidelines. Please close and contribute following our guidelines.

File: drivers/nfc/nrfxnfc/platform.c [hunk 1]
```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
```
Comment (carlescufi) [inline drivers/nfc/nrfxnfc/platform.c:4]: You cannot include source code with the Nordic 5-clause license in Zephyr. This license is not approved by the OSI and thus it is not a candidate for inclusion.

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Comment by HesselM** [comment]
> This pull request has been marked as stale because it has been open (more than) 60 days with no activity. Remove the stale label or add a comment saying that you would like to have the label removed otherwise this pull request will automatically be closed in 14 days. Note, that you can always re-open a closed pull request at any time.

I haven't got time to patch the requests yet. I guess it should be possible to keep the nrfxlib libraries as-is, including them from their original repo's and only include a vanilla-zephyr NFC driver for it (as is shown in this PR). That way this PR does not include the (nordic) binaries. As for the TNEP headers and files I have to check for possibilities: either drop it and only support raw-access to the NFC memory or retrieve it from the nrf-modules.

**Comment by nordicjm** [comment]
> HesselM changed the base branch from v3.5-branch to v3.4-branch

Not really sure what you are doing here but we do not accept new features for previous releases, if you want to add a new feature it must go into `main`

**Comment by HesselM** [comment]
Sorry for the mixup, i changed base which triggered a lot of automated stuff. 

As requested: the nordic files are removed and the proposed driver is formatted properly.

File: samples/drivers/nfc/prj.conf [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+CONFIG_NFC=y
+CONFIG_NFC_NDEF=y
+CONFIG_NFC_NDEF_MSG=y
+CONFIG_NFC_NDEF_RECORD=y
+CONFIG_NFC_NDEF_TEXT_RECORD=y
+CONFIG_NFC_NDEF_URI_REC=y
+CONFIG_NFC_NDEF_URI_MSG=y
+
+CONFIG_NFC_T2T_PARSER=y
```
Comment (nordicjm) [inline samples/drivers/nfc/prj.conf:19]: And as said before, zephyr does not accept code from ncs

**Review by nordicjm — CHANGES_REQUESTED** [review]

**Comment by HesselM** [comment]
@nordicjm : thanks, i pushed all updates to the proper (main) branch. Sorry for all these automated triggers. 

The KConfig you refer is a sample I adapted for the (generic) NFC driver and included it. It does depends on NCS libraries, but out-of-source, as stated in the readme.rst. I can also rework the sample to not require it all.

**Pull request closed without merging**